### PR TITLE
Add transcodeStatusUpdated

### PR DESCRIPTION
**Pull request by sittitep**
https://github.com/iterateco/iterate-media/issues/22

### CODE REVIEW

File: services/iterate-media/lib/handlers/transcode/statusUpdated.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+const env = require('../../env');
+const Helper = require('../../helper');
+const Media = require('../../services/media');
+
+function statusUpdated(event, context, callback) {
+  const message = event.Records[0].Sns.Message
+  const keys = message.outputKeyPrefix.split("/")
+  const appId = keys[0]
+  const mediaId = keys[1]
+
```
Comment (jetako) [inline services/iterate-media/lib/handlers/transcode/statusUpdated.js]: If the state is 'ERROR',  Call `console.error(message)`. Do this before calling `Media.update()`.

File: services/iterate-media/lib/handlers/transcode/statusUpdated.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  return Media.update(appId, mediaId, {status: message.state.toLowerCase()})
```
Comment (jetako) [inline services/iterate-media/lib/handlers/transcode/statusUpdated.js]: Only update the Media if the state is 'ERROR' or 'COMPLETED'.

File: services/iterate-media/lib/handlers/transcode/statusUpdated.js (hunk 1 above)
Comment (jetako) [inline services/iterate-media/lib/handlers/transcode/statusUpdated.js]: (New requirement) Also if the state is 'WARNING',  Call `console.warn(message)`

File: services/iterate-media/test/transcodeStatusUpdated.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    });
+  });
+
+  it('sns returns error', () => {
+    return wrapped.run(testHelper.SnsEvent({
+      outputKeyPrefix: `${DATA.MEDIA_1.appId}/${DATA.MEDIA_1.id}/`,
+      state: "ERROR"
+    })).then(response => {
+      assert.equal(response, 'success');
+      return Media.get(DATA.APP_1.id, DATA.MEDIA_1.id)
```
Comment (sittitep) [inline services/iterate-media/test/transcodeStatusUpdated.js]: @jetako  I could not figure it out why it's keep telling me "ItemNotFoundError: Item not found". It seems like `DATA.MEDIA_1` has never been created.

File: services/iterate-media/test/fixtures/data.js [hunk 4]
```diff
@@ -4,10 +4,20 @@ const TS_2 = 1479798305400;
...
   api_key: 'v0PfCBQlhvraWHZ9',
   name: 'App 1',
   created_at: TS_1,
   updated_at: TS_1
 };
+
+exports.APP_1 = APP_1;
+
+exports.MEDIA_1 = {
+  id: '1',
```
Comment (jetako) [inline services/iterate-media/test/fixtures/data.js]: In dynamodb, the Media `id` is a composite of the application id and media id. Should be:
```
id: `${APP_1.id}.1`
```

File: services/iterate-media/test/transcodeStatusUpdated.js (hunk 3 above)
Comment (jetako) [inline services/iterate-media/test/transcodeStatusUpdated.js]: See above comment: https://github.com/iterateco/iterate-media/pull/25/files/ad0385adb16b189515c8960887de277feb385246#r115348103

You'll need to change `Media.get(DATA.APP_1.id, DATA.MEDIA_1.id)` to `Media.get(DATA.APP_1.id, '1')`

**Comment by sittitep** [comment]
Tests are green now

**Pull request merged**